### PR TITLE
remove batch norm from StaticBuildBlackList

### PR DESCRIPTION
**Pull request by AndSonder**
### PR types

Others

### PR changes

Others

### Description

remove batch norm from StaticBuildBlackList

- 将 batch_norm 从 StaticBuildBlackList 中移除
- 修复 batch_norm 的注册信息，添加参数跳过fakeinit的相关判断
- 将 quantize_linear 迁移到phi下并补全注册信息
- 将 fused_bn_add_activation 迁移到phi下并补全注册信息
- 为 flatten_contiguous_range 添加参数跳过fakeinit的相关判断

相关issue:
- https://github.com/PaddlePaddle/Paddle/issues/55299

### CODE REVIEW

File: paddle/phi/ops/compat/fused_bn_add_activation_sig.cc [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (From00) [inline paddle/phi/ops/compat/fused_bn_add_activation_sig.cc]: 2022->2023

**Review by From00 — APPROVED** [review]
LGTM

**Review by phlrain — APPROVED** [review]
